src/communication_graph.jl

```diff
@@ -0,0 +1,88 @@
+
+function srcdest_to_rankarray(srcdest, rank)
+    if srcdest in ["all", "each", "some"]
+        return vcat(collect(0:getcommsize()-1))
+    end
+    if typeof(srcdest) <: Integer
+        return [srcdest]
+    end
+    if srcdest == nothing
+        return Int[]
+    end
+    return srcdest
+end
+
+struct MPIEventNeighbors
+    open_srcs::Vector{Int}
+    open_dst::Vector{Int}
+end
+
+function MPIEventNeighbors(ev::MPIEvent)
+    srcdest = getsrcdest(ev)
+    if srcdest == nothing
+        srcdest = (src = nothing, dest = nothing)
+    end
+    opensrcs = srcdest_to_rankarray(srcdest[:src], ev.rank)
+    opendsts = srcdest_to_rankarray(srcdest[:dest], ev.rank)
```

Comment:
Würde vielleicht `opendsts` in `opendests` umbenennen, damit wir einheitlich source als src und destination als dest abkürzen.